include/boost/url/detail/pct_encoding.hpp

```diff
@@ -44,6 +34,25 @@ key_equal_encoded(
     string_view plain_key,
     string_view encoded) noexcept;
 
+BOOST_URL_DECL
+int
+pct_decode_compare_unchecked(
+    string_view lhs,
+    string_view rhs) noexcept;
+
+BOOST_URL_DECL
+int
+pct_decode_icompare_unchecked(
+    string_view lhs,
+    string_view rhs) noexcept;
+
+BOOST_URL_DECL
+int
+icompare(
```

Comment:
Do these functions need to be exported? Only if they are called from an inline function. Otherwise they are internal to the shared library.

Reply:
I don't know. I was following the pattern from the `key_equal_encoded` function above. Should I remove `BOOST_URL_DECL` from all of them?

Reply:
you should find out where they are called and determine whether they should be exported or not